File: workflowtrigger/util/cluster/cluster_schema.go
package cluster

type ClusterSchema struct {
	ClusterIP   string        `json:"clusterIP"`
	ExposedPort int           `json:"exposedPort"`
	ClusterName string        `json:"clusterName"`
	NoDeploy    *NoDeployInfo `json:"noDeploy"`
}

type NoDeployInfo struct {
	Name      string `json:"name"`
	Reason    string `json:"reason"`
	Namespace string `json:"namespace"`
}

File: util/src/main/java/com/greenops/util/datamodel/cluster/ClusterSchema.java
package com.greenops.util.datamodel.cluster;


public class ClusterSchema {
    private String clusterIP;
    private int exposedPort;
    private String clusterName;
    private NoDeployInfo noDeployInfo;

    public ClusterSchema(String clusterIP, int exposedPort, String clusterName, NoDeployInfo noDeployInfo) {
        this.clusterIP = clusterIP;
        this.exposedPort = exposedPort;
        this.clusterName = clusterName;
        this.noDeployInfo = noDeployInfo;
    }

    public String getClusterIP() {
        return clusterIP;
    }

    public int getExposedPort() {
        return exposedPort;
    }

    public String getClusterName() {
        return clusterName;
    }

    public void setClusterIP(String clusterIP) {
        this.clusterIP = clusterIP;
    }

    public void setClusterName(String clusterName) {
        this.clusterName = clusterName;
    }

    public NoDeployInfo getNoDeployInfo() {
        return noDeployInfo;
    }
}

